src/parser/lexer.rs

```diff
@@ -272,8 +266,10 @@ fn lex_identifier_raw(input: &str) -> IResult<&str, String> {
     )(input)
 }
 
-fn lex_identifier(input: &str) -> LexResult {
-    lex_identifier_raw(input).map(|(input, ident)| (input, Token::Identifier(ident)))
+fn lex_command_or_identifier(input: &str) -> LexResult {
+    let (input, identifier) = lex_identifier_raw(input)?;
+    let token = recognize_command_or_identifier(identifier);
+    Ok((input, token))
```

Comment:
```suggestion
    let (remainder, identifier) = lex_identifier_raw(input)?;
    let token = recognize_command_or_identifier(identifier);
    Ok((remainder, token))
```

Just assuming that that's what that function is returning because of how nom usually works. More obvious to me if that's the case (vs returning the same thing as the input).

Reply:
Yup, see my response on #48 - but tl;dr not a bad idea but I'll favor consistency for the time being, and `input` is used consistently across the parser here.